javagen/src/main/java/com/azure/autorest/model/javamodel/JavaFile.java

```diff
@@ -84,7 +84,7 @@ public final void declareImport(List<String> imports) {
             // Only import paths that don't start with this file's package, or if they do start
             // with this file's package, then they must exist within a subpackage.
             imports = imports.stream()
-                    .filter(import_Keyword -> !import_Keyword.startsWith(package_Keyword)
+                    .filter(import_Keyword -> !import_Keyword.equals(package_Keyword)
```

Comment:
@srnagar Any idea why this get changed?

I've got lots of unnecessary import (import of files in exact same package) for new code generated.

E.g.

In `com.azure.management.resources.AliasType` import `com.azure.management.resources.AliasPathType`.

Reply:
Maybe this line got problem before `import_Keyword.indexOf('.', packageWithPeriodLength) != -1`?

Reply:
@weidongxu-microsoft I see the issue here. I was going to `com.azure.storage` classes to be able to import `com.azure.storage.implementation`. So, `startsWith` also doesn't work. I will add a fix to this issue. Thanks for reporting!